pypit/scripts/test_telluric_correction.py

```diff
@@ -0,0 +1,194 @@
+import sys
+from copy import deepcopy
+import numpy as np
+from numpy.polynomial.chebyshev import chebfit, chebval
+#import scipy.stats as st
+from scipy.optimize import fmin_cg, minimize
+import matplotlib.pyplot as plt
+from astropy.convolution import Gaussian1DKernel, convolve
+#import emcee
+from linetools.spectra.xspectrum1d import XSpectrum1D
+from pypit import arload
+
+
+# Data preperation functions:
+#   get_lrisr_resolution
+#   get_data
+#   get_kernel
+#   get_transimission
+
+# Fitting functions:
+#   get_template
+#   lnlike_tf
+#   opposite_lnlike_tf
+#   get_fscale
+#   tcorrect_data
+
+def get_lrisr_resolution(wavelength):
+    """
+    This comes from P. van Dokkum's fit
+    to the sky lines
+    """
+
+    l9   = (wavelength - 9000.)/1000.
+    resr = 1.26-0.128*l9+0.168*l9**2+0.1173*l9**3
+    resr = resr/wavelength*3.E5 # km s^-1
+
+    return np.array(resr)
+
+def get_data(fname):
+
+    tmp = arload.load_1dspec(fname, exten=1)
+
+    data = np.zeros((len(tmp.wavelength.value), 3))
+    data[:,0] = tmp.wavelength.value
+    data[:,1] = tmp.flux.value
+    data[:,2] = tmp.sig.value
+
+    return data
+
+def get_kernel(sp, inval=10000., outval=150):
+    clight = 299792.458
+
+    # Get FWHM of spectrum we're smoothing
+    # (assuming resolution is in R)
+
+    fwhm_in_ = sp.wavelength.value/inval
+    psize    = np.diff(sp.wavelength.value)
+    psize    = np.append(psize, sp.wavelength[-1].value - sp.wavelength[-2].value)
+
+    # Get FWHM of resolution we're smooth too
+    # (assuming resolution is in km/s)
+
+    fwhm_out = outval*2.*np.sqrt(2*np.log(2))
+    fwhm_out = (fwhm_out / clight) * sp.wavelength.value
+
+    fwhm_diff = (fwhm_out - fwhm_in_)/psize
+
+    return fwhm_diff
+
+def get_transmission(data):
+    """
+    This transmission spectrum comes from the ESO SkyCalc,
+    with default inputs.
+    """
+
+    tmp = np.loadtxt('pieter_transmission.dat')
+
+    # Convert transmission spectrum to angstroms
+    # and clip it to wavelength range of data
+    i = ((tmp[:,0]*10 >= min(data[:,0])) &
+         (tmp[:,0]*10 <= max(data[:,0])))
+    trans = XSpectrum1D.from_tuple((tmp[:,0][i]*10, tmp[:,1][i]))
+
+    # Convolve the atm. transsissiom spectrum to LRIS_R resolution
+    res         = get_lrisr_resolution(trans.wavelength.value)
+    fwhm_pix    = get_kernel(trans, inval=20000., outval=res)
+
+    smooth_spec = np.zeros((len(trans.wavelength.value)))
+    for i, fp in enumerate(fwhm_pix):
```

Comment:
This looks painfully slow.  
But maybe we are forced to something like
this for a wavelength dependent FWHM.
Let's discuss.